department-of-veterans-affairs/va.gov-cms
# VACMS-15559 Migration of service location data to paragraphs

## Description

Relates to #15559

## Testing done


## Screenshots


## QA steps

What needs to be checked to prove this works?
What needs to be checked to prove it didn't break any related things?
What variations of circumstances (users, actions, values) need to be checked?

As admin
1. Go to the [tugboat terminal](https://tugboat.vfs.va.gov/65a9a66f067b1809aa202f70)
   - [x] enter `drush scr ./scripts/content/VACMS-15559-migrate-service-location-from-node-to-paragraph.php`
   - [x] You should see it progressing in groups of 25.     
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/5752113/576da597-f3ee-4488-bfd2-fdbff9026cc3)
   - [ ] Revisit it ~2hrs later
   - [ ] You should see the following message.  (the numbers might differ slightly based on current content.)
       ![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/5752113/287e9c58-92e0-4952-8c7b-3e4242f4ba52)

2. Facility service with multiple service locations https://pr16911-frzioxcuceuh23pjuqcftuoyfqqthsg3.ci.cms.va.gov/pittsburgh-health-care/locations/pittsburgh-va-medical-center-university-drive/gastroenterology
   - [x] Validate that under "Recent changes" you see 1 entry at the top from CMS Migrator indicating the node was updated by an "Automated move ..." 
   - [x] Click the edit tab and Validate that each of the two "Service Locations" have the same "Appointment phone" that (numbers, type and label) that is found under the "Scheduling" section.
   - [x] Click the revision tab and validate the revision log is there as well and that there is only one of them.
   - [x] Visit the /admin/content/audit/service-locations-detailed?title=Gastroenterology&type_op=in&type%5B%5D=health_care_local_health_service&moderation_state_multiple_workflow_filter=All&field_administration_target_id=12
      - [x]  Validate old and new apt text match
      - [x] Validate Apt lead-in and Apt custom match
      - [x] validate new and old online scheduling match
      - [x] validate one apt phone became 2(because there are two locations) and they match
      - [x] validate "use facility phone" is off, because a phone was provided.
      - [x] validate walkins: "unspecified" became "Yes by apt only"
3. Custom apt text  Visit [the filtered audit](https://pr16911-frzioxcuceuh23pjuqcftuoyfqqthsg3.ci.cms.va.gov/admin/content/audit/service-locations-detailed?title=intimate&type_op=in&moderation_state_multiple_workflow_filter=All&field_administration_target_id=12)
   - [x] Validate that for services that have "Old apt leadin text" have matching "New apt custom text"
4. Multiple Apt phone numbers & phone with custom labels: Visit [the filtered View](https://pr16911-frzioxcuceuh23pjuqcftuoyfqqthsg3.ci.cms.va.gov/admin/content/audit/service-locations-detailed?title=women&type_op=in&moderation_state_multiple_workflow_filter=All&field_administration_target_id=12), and scroll down to the bottom  for "Women Veteran care - Pittsburgh VA Medical Center-University Drive"
   - [x] Validate that both old and new apt phone numbers have two entries
   - [x] Validate that custom labels they have match old and new.
5. Double check that forward revisions were handled correctly by going to https://pr16911-frzioxcuceuh23pjuqcftuoyfqqthsg3.ci.cms.va.gov/node/5567
   - [x] Validate that there are two revisions that were added by this migration
   - [x] Validate that the top most revision is the draft revision.

### Definition of Done


### Select Team for PR review



### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
